src/meili/map_matcher.cc

```diff
@@ -416,9 +419,16 @@ MapMatcher::OfflineMatch(const std::vector<Measurement>& measurements, uint32_t
       const auto& interpolated_results = InterpolateMeasurements(*this, it->second, this_stateid, next_stateid);
 
       // Copy the interpolated match results into the final set
-      std::copy(interpolated_results.cbegin(), interpolated_results.cend(), std::back_inserter(best_paths.back()));
+      std::copy(interpolated_results.cbegin(), interpolated_results.cend(), std::back_inserter(path));
     }
 
+    // Keep the path and the cost for it, if you cant find a valid state id we just give it a huge cost
+    auto found = std::find_if(stateids.rbegin(), stateids.rend(), [](const StateId& si) {
+      return si.IsValid();
+    });
+    auto accumulated_cost = found != stateids.rend() ? vs_.AccumulatedCost(*found) : MAX_ACCUMULATED_COST;
```

Comment:
@ptpt i think ive properly gotten the last valid stateids accumulated cost now. what are your thoughts on handling discontinuities. i know i had mentioned adding cost where those occured. do you think that would be best done here or perhaps...